pkg/collect/host_kernel_configs.go

```diff
@@ -0,0 +1,142 @@
+package collect
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+)
+
+type CollectHostKernelConfigs struct {
+	hostCollector *troubleshootv1beta2.HostKernelConfigs
+	BundlePath    string
+}
+
+type KConfigs map[string]string
+
+const HostKernelConfigsPath = `host-collectors/system/kernel-configs.json`
+
+const (
+	kConfigBuiltIn  string = "y"
+	kConfigAsModule string = "m"
+	kConfigLeftOut  string = "n"
+)
+
+func (c *CollectHostKernelConfigs) Title() string {
+	return hostCollectorTitleOrDefault(c.hostCollector.HostCollectorMeta, "kernel-configs")
+}
+
+func (c *CollectHostKernelConfigs) IsExcluded() (bool, error) {
+	return isExcluded(c.hostCollector.Exclude)
+}
+
+func (c *CollectHostKernelConfigs) Collect(progressChan chan<- interface{}) (map[string][]byte, error) {
+
+	kernelRelease, err := getKernelRelease()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get kernel release")
+	}
+
+	var kConfigs KConfigs
+	kConfigs, err = loadKConfigs(kernelRelease)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load kernel configs")
+	}
+
+	b, err := json.Marshal(kConfigs)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal kernel configs")
+	}
+
+	output := NewResult()
+	output.SaveResult(c.BundlePath, HostKernelConfigsPath, bytes.NewBuffer(b))
+
+	return output, nil
+}
+
+func getKernelRelease() (string, error) {
+	out, err := exec.Command("uname", "-r").Output()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to determine kernel release")
```

Comment:
maybe add more details like
failed to determine kernel release using 'uname -r'

Reply:
many thanks Dexter I have udpated the message